H_ecto_models.md

```diff
@@ -597,6 +597,73 @@ end
+defmodule HelloPhoenix.Video do
+. . .
+  before_update :reset_likes_count
+  def reset_likes_count(changeset) do
+    changeset
+    |> Ecto.Changeset.put_change(:likes, 0)
+  end
+end
+```
+Now, when `Repo.update` is called for the `Video` model, the `likes` field will automatically be reset to zero. This should keep us safe from the ravages of an often hostile modern Internet!
+
+`Ecto.Model.Callbacks` actually ships with with a lot more than the `before_update` callback, in addition, it comes with:
+
+- `before_delete`: called before the adapter deletes our record from the database; if perhaps we have a foreign_key constraint from another table, we would use this hook to clean it up.
+- `after_delete`: called after the adapter deletes our record. If we wished to archive our record, we'd do it in this hook. 
+- `before_update`: called before the adapter updates our record. We used it above to reset the likes counter.
+- `after_update`: called after the adapter updates our record. If we wanted to update our user that his model has changed, now would do that here.
+- `before_insert`: called before the adapter first creates a new record. We would use it in cases when we wanted to infer default values to certain fields from the partial changeset we receive from the user.
+- `after_insert`: called after the adapter creates our record. We could use this to notify the user a new has been created under his account.
+- `after_load`: called after the model is loaded from the database. We might use this to load in other values based upon what we found in our record.
+
+Note: there is no `before_load` hook.
+
+In all cases, a hook expects to be passed an atom that is the name of the function it is to call if that function is defined in the current module:
+
+```elixir
+before_insert :name_of_function_to_call
+def name_of_function_to_call(changeset) do
+  # do stuff with the changeset
+  # and return the modified changeset
+end
+```
+If our callback function lives elsewhere, hooks can take the name of the module as the first argument:
+```elixir
+before_insert HelloPhoenix.AnotherPlace, :name_of_function_to_call
+```
+
+Our callback function must always take a changeset as its first argument, and return a changeset. In our callbacks, we can expect to often be calling `Ecto.Changeset` functions like:
+
+- `put_change(changeset, key, value)`
+- `get_field(changeset, key, defaultValue//nil)`
+- `add_error(changeset, key, error)`
+
+For the full reference, visit the docs at [Ecto Changeset documentation](http://hexdocs.pm/ecto/Ecto.Changeset.html).
+
+Finally, a word to the wise, callbacks and hooks give us great power to manipulate our business logic, but with great power comes great responsibility... and misusing that power will lead to an extremely messy code-base that's difficult to debug / test / develop (and lead to the death of your uncle). We should avoid stacking tons of IO-related "side-effects" into our callbacks. So avoid stuff like the following:
```

Comment:
I think this is an important paragraph, so I want to make sure that we're as simple an clear as possible. 

I took a first pass at an edit, and I'm curious what you think:
Ecto callbacks give can be very powerful, but we need to take care when using them. On one hand, they can add behaviors in a way which may not be apparent in the execution flow because callbacks can be defined far from the action in our model. On the other hand, it might seem tempting to put a lot of behavior in a single callback, which can make the model difficult to test and debug.